File: backend/models/broker_event.go
//
// Date: 2018-09-11
// Author: Spicer Matthews (spicer@cloudmanic.com)
// Last Modified by: Spicer Matthews
// Last Modified: 2018-09-14
// Copyright: 2017 Cloudmanic Labs, LLC. All rights reserved.
//

package models

import (
	"time"
)

type BrokerEvent struct {
	Id              uint      `gorm:"primary_key" json:"id"`
	UserId          uint      `sql:"not null;index:UserId" json:"_"`
	CreatedAt       time.Time `json:"_"`
	UpdatedAt       time.Time `json:"_"`
	BrokerAccountId uint      `sql:"not null;index:BrokerAccountId" json:"_"`
	BrokerId        string    `json:"_"`
	Type            string    `sql:"not null;type:ENUM('Ach', 'Trade', 'Option', 'Interest', 'Journal', 'Dividend', 'Adjustment', 'Other');default:'Other'" json:"type"`
	Date            Date      `gorm:"type:date" json:"date"`
	Amount          float64   `json:"amount"`
	Symbol          string    `json:"symbol"`
	Commission      float64   `json:"commission"`
	Description     string    `json:"description"`
	Price           float64   `json:"price"`
	Quantity        int64     `json:"quantity"`
	TradeType       string    `sql:"not null;type:ENUM('Equity', 'Option', 'ETF', 'Preferred Stock', 'Other');default:'Other'" json:"trade_type"`
}

/* End File */

File: frontend/src/app/models/broker-event.ts
//
// Date: 9/12/2018
// Author(s): Spicer Matthews (spicer@options.cafe)
// Copyright: 2018 Cloudmanic Labs, LLC. All rights reserved.
//

import * as moment from 'moment';

export class BrokerEvent 
{
  public Id: number;
  public Date: Date;
  public Type: string;
  public Amount: number;
  public Symbol: string;
  public Commission: number;
  public Description: string;
  public Price: number;
  public Quantity: number;
  public TradeType: string;

  //
  // Build from JSON list.
  //
  fromJsonList(json: Object[]): BrokerEvent[] {
    let result = [];

    if (!json) 
    {
      return result;
    }

    for (let i = 0; i < json.length; i++) 
    {
      result.push(new BrokerEvent().fromJson(json[i]));
    }

    // Return happy
    return result;
  }

  //
  // Json to Object.
  //
  fromJson(json: Object): BrokerEvent {
    let obj = new BrokerEvent();

    obj.Id = json["id"];
    obj.Date = moment(json["date"]).toDate();
    obj.Type = json["type"];
    obj.Amount = json["amount"];
    obj.Symbol = json["symbol"];
    obj.Commission = json["commission"];
    obj.Price = json["price"];
    obj.Quantity = json["quantity"];
    obj.TradeType = json["trade_type"];
    obj.Description = json["description"];

    return obj;
  }

  
}

/* End File */